liesel/model/model.py

```diff
@@ -214,7 +214,13 @@ def _all_nodes_and_vars(self) -> tuple[list[Node], list[Var]]:
         (as recursive inputs) added to the graph.
         """
         nodes = self.nodes.copy()
-        nodes.extend(node for var in self.vars for node in var.nodes)
+        # nodes.extend(node for var in self.vars for node in var.nodes)
+
+        for var in self.vars:
+            for node in var.nodes:
+                if node not in nodes:
+                    nodes.append(node)
+
```

Comment:
What does this change accomplish? Isn't it pretty much the same as before? Why is line 217 commented out instead of deleted?

Reply:
Right. Its is useless. I removed it